Add allowlist note for TFE users. (#2038)
hashicorp/terraform-website: content/cloud-docs/registry/add.mdx

You can add providers and modules from the public [Terraform Registry](/registry) to your Terraform Cloud private registry. This lets you clearly designate which public providers and modules are recommended for the organization and makes their supporting documentation and examples centrally accessible.

> -> **Note:** Public providers and modules are available in Terraform Enterprise v202012-1 and later. You will need to make sure access to registry.terraform.io and  https://yy0ffni7mf-dsn.algolia.net/ are allowed.

Reviewer: What about something like this? Just trying to strengthen the wording a bit!

```suggestion
-> **Note:** Public providers and modules are available in Terraform Enterprise v202012-1 and later. Your instance must allow access to registry.terraform.io and https://yy0ffni7mf-dsn.algolia.net/.
```
